allow '=' in values of pandoc key=value pairs

kevinushey opened the pull request:
Closes https://github.com/quarto-dev/quarto/issues/813.

cscheid commented:
Thanks for the PR!

We need to be super careful with stuff like this - not that pandoc_attr.ts is particularly careful, but I don't want to make situation worse. I think I'd want some tests around this behavior so we don't regress on it, but also understand the blast radius of the change.

vezwork commented on packages/editor/src/api/pandoc_attr.ts:
```diff
@@ -311,8 +311,10 @@ export function pandocAttrKeyvalueFromText(text: string, separator: ' ' | '\n'):
   return lines.map(line => {
-    const parts = line.trim().split('=');
-    return [parts[0], (parts[1] || '').replace(/^"/, '').replace(/"$/, '')];
+    const idx = line.indexOf('=');
```
yea I suppose so! like `if (idx === -1) return line; else { bla }`

vezwork reviewed (approved):
`attr-equals` is quite extensive! Nice. Perhaps there should also be some lines of prose/markdown that contain equals symbols?

Also, could you add a description of the fix to https://github.com/quarto-dev/quarto/blob/main/apps/vscode/CHANGELOG.md please?

Looks great otherwise! Excited to have this fix!

kevinushey commented:
One thing I'm missing -- how does `yarn test-vscode` find Quarto? Is it expected to just be available on the PATH?

kevinushey commented:
Thanks! I updated the change log, and added a couple more "weird" examples for the roundtrip test. I don't have permissions to merge here so I'll ask you to do so if everything looks good.
